rust/rope/src/rope.rs

```diff
@@ -633,15 +637,42 @@ impl Rope {
 
     // TODO: this should be a Cow
     // TODO: a case can be made to hang this on Cursor instead
-    pub fn slice_to_string(&self, start: usize, end: usize) -> String {
+    pub fn slice_to_string<T>(&self, range: T) -> String 
+        where T: RangeBounds<usize>
+    {
         let mut result = String::new();
-        for chunk in self.iter_chunks(start, end) {
+        for chunk in self.iter_chunks(range) {
             result.push_str(chunk);
         }
         result
     }
 }
 
+// Helper methods for RangeBounds implementation
```

Comment:
This is what I'm a little concerned about. I hope it's idiomatic enough. Would love suggestions if it doesn't seem to fit in right.

Reply:
I think this is fine. As mentioned before, you could also consider having a single fn that returns both start and end.